File: algorithms/0476.number-complement/solution.rb
# @param {Integer} num
# @return {Integer}
def find_complement(num)
  # num.to_s(2).chars.map {|c| c == '0' ? '1' : '0' }.join.to_i(2)

  num.to_s(2).tr('01', '10').to_i(2)
end

File: algorithms/0476.number-complement/solution.rs
impl Solution {
    pub fn find_complement(num: i32) -> i32 {
        let s = format!("{:b}", num).to_string();

        let mut res = 0i32;
        for (i, c) in s.chars().rev().enumerate() {
            if c == '0' {
                res += 2i32.pow(i as u32);
            }
        }

        res
    }
}
